### ISSUE TITLE
Enhancement: utf8mb4_bin - Emoji Support

### ISSUE THREAD

**Issue by StirCrazyStu** [mybb/mybb#3607]
Emoji Support

Can the MyBB database collation for the Posts table be utf8mb4_bin, by default (and converted on the next upgrade)?

Ta :)

**Comment by woostarUK** [comment]
Ignore this 🤦‍♂️🤦‍♂️ I hadn't changed the option in config.php

Although, I still think the post table should be utf8mb4 by default :)

**Comment by Ben-MyBB** [comment]
@mybb/developers

**Comment by Eldenroot** [comment]
I agree, it should not have any side effect and could bring new possibilities to users.

**Comment by WildcardSearch** [comment]
I agree, as well.

I changed my board over a while back and there have been no unwanted side effects at all. My mobile users appreciate the ability to use their emojis.

**Comment by Sama34** [comment]
Unless I missed the discussion somewhere else, this seems valid for at least 1.9, so I will open back.

Regards.

**Comment by arcalys** [comment]
How should this be handled for the other DBMS?
I don’t have much experience with SQLite, so I can’t speak for it.
In PostgreSQL, there is no `utf8mb4_bin`. UTF8 in PostgreSQL fully supports all Unicode characters (1–4 bytes) but to replicate MySQL’s binary-safe behavior, we would need to set a binary collation on affected columns.
